packages/amplify-storage-simulator/src/__tests__/S3server.test.ts

```diff
@@ -169,10 +176,37 @@ describe("Test put api", () => {
   test(" multipart upload", async () => {
     const params = {
       Bucket: bucket, // pass your bucket name
-      Key: "upload/long_image.jpg", 
+      Key: "upload/long_image.jpg",
       Body: buf2
     };
     const data = await s3client.upload(params).promise();
     expect(data.Key).toBe("upload/long_image.jpg");
   });
+
+  test(" async uploads", async () => {
+    const params1 = {
+      Bucket: bucket, // pass your bucket name
+      Key: "upload/long_image1.jpg",
+      Body: buf2
+    };
+    const data = await s3client.upload(params1).promise();
```

Comment:
You should add this to an array instead of awaiting for promise. If you awiat the other uploads will happen only after this one is complete
```typescript
const uploadPromises = [ ];
uploadPromises.push(s3client.upload(params1).promise());
uploadPromises.push(s3client.upload(params2).promise());
uploadPromises.push(s3client.upload(params3).promise());

const uploadResults = await Promise.all(uploadPromises);

expect(uploadResults[0].data.Key).toBe('uploda/long_image1.jpg');
...
...

Reply:
changed